src/iop/expfusion.c

```diff
@@ -0,0 +1,1268 @@
+                                      const float exposure_optimum, const float exposure_width,
+                                      const float exposure_left_cutoff, const float exposure_right_cutoff)
+{
+  if((exposure_left_cutoff > 0.f && lum < exposure_left_cutoff) || (exposure_right_cutoff < 1.f && lum > exposure_right_cutoff))
+    return 0.f;
+
+  float exp = 1.f;
+
+  const float v = (lum - exposure_optimum) / exposure_width;
+
+  if(weight_mode == DT_WEIGHT_MODE_GAUSSIAN)
+  {
+    exp = dt_fast_expf(-(v * v) * .5f);
+  }
+  else if(weight_mode == DT_WEIGHT_MODE_LORENTZIAN)
+  {
+    exp = 1.f / (1.f + ((v * v) * .5f));
+  }
+  else if(weight_mode == DT_WEIGHT_MODE_HALF_SIZE)
+  {
+    if(fabs(v) <= (float)M_PI * .5f)
+      exp = cos(v);
+    else
+      exp = 0.f;
+  }
+  else if(weight_mode == DT_WEIGHT_MODE_FULLSINE)
+  {
+    if(fabs(v) <= (float)M_PI)
+      exp = (1.f + cos(v)) * .5f;
+    else
+      exp = 0.f;
+  }
+  else if(weight_mode == DT_WEIGHT_MODE_BISQUARE)
+  {
+    if(fabs(v) <= 1.f)
+      exp = (1.f - powf(v, 4.f));
+    else
+      exp = 0.f;
+  }
+
+  return exp;
+}
+
+static void _buil_weight_map(const float *const img_src, const size_t wd, const size_t ht, const int ch,
+                             float *const img_map, const dt_iop_grey_projectors_t grey_projector, const dt_iop_weight_modes_t weight_mode,
+                             const float exposure_optimum, const float exposure_width,
+                             const float exposure_left_cutoff, const float exposure_right_cutoff,
+                             const dt_iop_order_iccprofile_info_t *const work_profile)
+{
+#ifdef _OPENMP
+#pragma omp parallel for default(none) \
+  dt_omp_firstprivate(img_src, img_map, ht, wd, ch, grey_projector, work_profile, weight_mode, \
+  exposure_optimum, exposure_width) \
+  schedule(static) \
+  collapse(2)
+#endif
+  for(size_t y = 0; y < ht; y++)
+  {
+    for(size_t x = 0; x < wd; x++)
+    {
+      const float *const rgb = img_src + (y * wd * ch) + x * ch;
+
+      float E = 1.f;
+      if(grey_projector != DT_PROJECTOR_NONE)
+      {
+        const float lum = _grey_projector(rgb, grey_projector, work_profile);
+        E = _well_exposedness(lum, weight_mode, exposure_optimum, exposure_width, exposure_left_cutoff, exposure_right_cutoff);
+      }
+      else
+      {
+        E = _well_exposedness(rgb[0], weight_mode, exposure_optimum, exposure_width, exposure_left_cutoff, exposure_right_cutoff) *
+              _well_exposedness(rgb[1], weight_mode, exposure_optimum, exposure_width, exposure_left_cutoff, exposure_right_cutoff) *
+              _well_exposedness(rgb[2], weight_mode, exposure_optimum, exposure_width, exposure_left_cutoff, exposure_right_cutoff);
+      }
+
+      img_map[y * wd + x] = E;
+    }
+  }
+}
+
+static float exposure_increment(const float stops, const int e)
+{
+  const float white = exp2f(-stops * (float)e);
+  const float scale = 1.0f / white;
```

Comment:
given that exp2f(-x) = 1/exp2f(x) - why do this as opposed to removing the - from the exp2f argument and returning that?